crates/uv/src/commands/python/install.rs

```diff
@@ -540,7 +638,19 @@ fn create_bin_links(
 
     for target in targets {
         let target = bin.join(target);
-        match installation.create_bin_link(&target) {
+        let executable = if upgradeable {
```

Comment:
When creating `bin` links, we now try to use a symlink directory as the `bin` link target if we can and this is an upgradeable request.